File: sample/src/main/java/ru/surfstudio/standard/interactor/common/response/ResultResponse.java
package ru.surfstudio.standard.interactor.common.response;

import com.google.gson.annotations.SerializedName;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Ответ от сервера содержащий результат
 */

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ResultResponse<T> extends ErrorResponse {

    @SerializedName("result")
    private T result;
}

File: template/src/main/java/ru/surfstudio/standard/interactor/common/response/ResultResponse.kt
package ru.surfstudio.standard.interactor.common.response

import com.google.gson.annotations.SerializedName

/**
 * Ответ от сервера содержащий результат
 */

class ResultResponse<T> : ErrorResponse() {

    @SerializedName("result")
    var result: T? = null
        set(result) {
            field = this.result
        }
}
